spec/unit/remove_task_spec.rb

```diff
@@ -47,12 +47,10 @@
       remove_task.remove(store, test_name, { :verbose => true })
     end
 
-    it "show a warning when SystemDescription does not exist" do
-      expect(Machinery::Ui).to receive(:warn).with(
-        "System description \"foo_bar_does_not_exist\" does not exist."
-      )
-
-      remove_task.remove(store, "foo_bar_does_not_exist")
+    it "throws an error when SystemDescription does not exist" do
+      expect {
+        remove_task.remove(store, "foo_bar_does_not_exist")
+      }.to raise_error(Machinery::Errors::SystemDescriptionNotFound)
```

Comment:
I would remove two descriptions of which is one is missing and make sure that the other one is removed besides showin the error to adhere to the acceptance criteria.